set.mm

```diff
@@ -735799,6 +735836,340 @@ their group (addition) operations are equal for all pairs of elements of
+       semigroup.  Formerly part of proof for ~ gsumccat .  (Contributed by AV,
+       26-Dec-2023.) $)
+    gsumsgrpccat $p |- ( ( G e. SGrp /\ ( W e. Word B /\ X e. Word B )
+                                     /\ ( W =/= (/) /\ X =/= (/) ) )
+               -> ( G gsum ( W ++ X ) ) = ( ( G gsum W ) .+ ( G gsum X ) ) ) $=
+      ( wcel wa cfv caddc co c1 cmin cc0 cseq wceq cn0 syl wf vx vy csgrp cword
+      vz c0 wne chash cconcat cgsu simp1 cmgm sgrpmgm mgmcl syl3an1 3expb sylan
+      w3a cv sgrpass cuz cn lennncl ad2ant2r 3adant1 nnzd uzidd nnm1nn0 uzaddcl
+      ad2ant2l syl2anc nncnd 1cnd addsubassd ax-1cn npcan sylancl 3eltr4d nn0uz
+      cc fveq2d syl6eleq cfz cfzo ccatcl 3ad2ant2 wrdf ccatlen oveq2d cz fzoval
+      zaddcld eqtrd feq2d mpbid seqsplit simpl2l simpl2r eleq2d biimpar syl3anc
+      ffvelrnda ccatval1 seqfveq addid2d eqtr4d seqeq1d addcomd addsubd fveq12d
+      oveq1d ccatval3 eqcomd seqshft2 oveq12d nnaddcld gsumval2 simp2l 3eqtr4d
+      simp2r ) CUCHZDAUDZHZEYBHZIZDUFUGZEUFUGZIZURZDUHJZEUHJZKLZMNLZBDEUILZOPZJ
+      ZYJMNLZBDOPJZYKMNLZBEOPJZBLZCYNUJLCDUJLZCEUJLZBLYIYPYQYOJZYMBYNYQMKLZPZJZ
+      BLUUAYIUAUBUEBAYNOYQYMYIYAUAUSZAHZUBUSZAHZIUUHUUJBLZAHZYAYEYHUKZYAUUIUUKU
+      UMYACULHUUIUUKUUMCUMACUUHUUJBFGUNUOUPUQYIYAUUIUUKUEUSZAHURUULUUOBLUUHUUJU
+      UOBLBLQUUNACUUHUUJBUUOFGUTUQYIYJYSKLZYJVAJZYMUUEVAJYIYJUUQHYSRHZUUPUUQHYI
+      YJYIYJYEYHYJVBHZYAYCYFUUSYDYGADVCVDVEZVFZVGYIYKVBHZUURYEYHUVBYAYDYGUVBYCY
+      FAEVCVJVEZYKVHSZYSYJYJVIVKYIYJYKMYIYJUUTVLZYIYKUVCVLZYIVMZVNYIUUEYJVAYIYJ
+      VTHMVTHUUEYJQUVEVOYJMVPVQZWAVRYIYQROVAJZYIUUSYQRHUUTYJVHSVSWBZYIOYMWCLZAU
+      UHYNYIOYNUHJZWDLZAYNTZUVKAYNTYIYNYBHZUVNYEYAUVOYHADEWEWFAYNWGSYIUVMUVKAYN
+      YIUVMOYLWDLZUVKYIUVLYLOWDYEYAUVLYLQYHADEWHWFWIYIYLWJHUVPUVKQYIYJYKUVAYIYK
+      UVCVFZWLOYLWKSWMWNWOZXBWPYIUUDYRUUGYTBYIBUAYNDOYQUVJYIUUHOYQWCLZHZIYCYDUU
+      HOYJWDLZHZUUHYNJUUHDJQYCYDYAYHUVTWQYCYDYAYHUVTWRYIUWBUVTYIUWAUVSUUHYIYJWJ
+      HUWAUVSQUVAOYJWKSZWSWTADEUUHXCXAXDYIUUGYSYJKLZBYNOYJKLZPZJYTYIYMUWDUUFUWF
+      YIUUEUWEBYNYIUUEYJUWEUVHYIYJUVEXEXFXGYIYMYKYJKLZMNLUWDYIYLUWGMNYIYJYKUVEU
+      VFXHXKYIYKYJMUVFUVEUVGXIWMXJYIBUAEYNYJOYSYIYSRUVIUVDVSWBZUVAYIUUHOYSWCLZH
+      ZIZUUHYJKLYNJZUUHEJZUWKYCYDUUHOYKWDLZHZUWLUWMQYCYDYAYHUWJWQYCYDYAYHUWJWRY
+      IUWOUWJYIUWNUWIUUHYIYKWJHUWNUWIQUVQOYKWKSZWSWTADEUUHXLXAXMXNXFXOWMYIABYNC
+      OYMUCFGUUNYIYMRUVIYIYLVBHYMRHYIYJYKUUTUVCXPYLVHSVSWBUVRXQYIUUBYRUUCYTBYIA
+      BDCOYQUCFGUUNUVJYIUWAADTZUVSADTYIYCUWQYAYCYDYHXRADWGSYIUWAUVSADUWCWNWOXQY
+      IABECOYSUCFGUUNUWHYIUWNAETZUWIAETYIYDUWRYAYCYDYHXTAEWGSYIUWNUWIAEUWPWNWOX
+      QXOXS $.
+  $}
+
+  ${
+    gsumccatNEW.b $e |- B = ( Base ` G ) $.
+    gsumccatNEW.p $e |- .+ = ( +g ` G ) $.
+    $( Homomorphic property of composites.  Second formula in [Lang] p. 4.
+       (Contributed by Stefan O'Rear, 16-Aug-2015.)  (Revised by Mario
+       Carneiro, 1-Oct-2015.)  (Proof shortened by AV, 26-Dec-2023.)
```

Comment:
As above, removing nonemptiness condition.

Reply:
Here (for monoids) are no nonemptiness conditions, but the cases for empty words must be handles separately (as in the original proof).

Reply:
That's why I wrote "removing the nonemptiness condition" !